eth/api.go

```diff
@@ -317,6 +319,24 @@ type BlockTraceResult struct {
 	Error      string                `json:"error"`
 }
 
+// TraceArgs holds extra parameters to trace functions - either a vm.LogConfig, or a
+// string to evaluate as a JavaScript object.
+type TraceArgs struct {
+	logConfig  *vm.LogConfig
+	tracer     string
+}
+
+func (ta *TraceArgs) UnmarshalJSON(data []byte) error {
+	input := strings.TrimSpace(string(data))
+
+	if len(input) > 2 && input[0] == '"' && input[len(input) - 1] == '"' {
```

Comment:
Perhaps this should be `len(input) >= 2`. Otherwise we get a weird error for the empty string:

```js
> debug.traceTransaction("0x9eb11d1e042a22661e23ef69300e32426a8448495edffdecf81564ab4f47bdb0", "")
json: cannot unmarshal string into Go value of type vm.LogConfig
    at web3.js:3119:20
    at web3.js:6023:15
    at web3.js:4995:36
    at <anonymous>:1:1
```